### PR TITLE
[v2.10] Bump k8s version to 1.31

### PR DESCRIPTION
**Pull request by gbuenodevsuse**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,44 +7,44 @@ toolchain go1.22.3
...
-	github.com/prometheus/client_model v0.5.0
-	github.com/prometheus/common v0.45.0
-	github.com/rancher/lasso v0.0.0-20240603075835-701e919d08b7
-	github.com/rancher/rancher/pkg/apis v0.0.0-20240613110601-c21147a009c6
-	github.com/rancher/wrangler/v3 v3.0.0-rc2
+	github.com/prometheus/client_model v0.6.1
+	github.com/prometheus/common v0.55.0
+	github.com/rancher/lasso v0.0.0-20240924233157-8f384efc8813
+	github.com/rancher/rancher/pkg/apis v0.0.0-20241001074650-6b63832fda0f
+	github.com/rancher/wrangler/v3 v3.0.0
```
Comment (mbelur) [inline go.mod]: Is there a specific reason to use v3.0.0 version of wrangler? I see this commit 
https://github.com/rancher/wrangler/commit/3e8181532ae254bf863dd8246ab711910d84f1c5 
which is not available in v3.0.0.

Comment (gbuenodevsuse) [reply go.mod]: I wasn't sure which version of wrangler to use, so I went with the latest released tag available. Should we be using v3.0.1-rc2 instead?

Comment (mbelur) [reply go.mod]: For now we could use v3.0.1-rc2 but I think it's best to the v3.0.1 once available.

**Review by guangyee — APPROVED** [review]
lgtm

**Pull request merged**